include/rocksdb/env.h

```diff
@@ -163,6 +163,33 @@ class Env {
   // The result of Default() belongs to rocksdb and must never be deleted.
   static Env* Default();
 
+  // Handles the event when a new DB or a new ColumnFamily starts using the
+  // specified data paths.
+  //
+  // The data paths might be shared by different DBs or ColumnFamilies,
+  // so OnDbPathsRegistered might be called with the same data paths.
+  // For example, when CreateColumnFamily is called multiple times with the same
+  // data path, OnDbPathsRegistered will also be called with the same data path.
+  //
+  // When a path is called in OnDbPathsRegistered, it must be correspondingly
+  // called in OnDbPathsUnregistered.
+  //
+  // Different implementations may take different actions.
+  // By default, it's a no-op.
```

Comment:
I think it is better to document here the behavior when Status returned is not OK. I assume what we want is that, if it is not OK, nothing should change and users don't have to call UnregisterDbPaths() to unregister them?